### PR TITLE
Added Tree Select

### PR DESCRIPTION
**Pull request by kevde**

### CODE REVIEW

**Comment by mattmonihan** [comment]
Just started reviewing :)

File: src/fields/ReactTreeSelectField.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    });
+    return selectedOptions;
+  }
+
+  handleChange = (valueOptions) => {
+    const value = valueOptions.map(option => option.value);
+    this.props.onChange(value); 
+  }
+
+  componentDidMount() {
```
Comment (mattmonihan) [inline src/fields/ReactTreeSelectField.js:29]: @kevde Is this left over from the signature component? If so, you can remove this function.

**Review by mattmonihan — COMMENTED** [review]
This works on my end.

In the issue: #16 there's a Phase 2 comment about loading the treeOptions through a fetch. Is it straightforward to add that? Perhaps we'd pass in the options like:

```js
 "ui:options": {

  remote: {
     data: [
       {
         id: 1,
         url: "https://domain.com/path",
         record: ["items"],
         label: ["label_name"],
         value: ["value_object", "value"]
       },
       {
         id: 2,
         parent: 1,
         url: "https://domain.com/path/${parent}",
         record: ["items"],
         label: ["label_name"],
         value: ["value_object", "value"]
       }
    ]
  }
}

```

**Comment by kevde** [comment]
Is the value really an array per item? will do for this week

**Comment by mattmonihan** [comment]
Oh, I should explain why I had:
```js
{
  record: ["items"],
  label: ["label_name"],
  value: ["value_object", "value"]
}
```

Sometimes, when an API returns a json response, the format it's in is not the best for loading into the picker.

For example, you might get back a response for some api like:

```js
{
  people: [
    {
      name: "matt",
      address: {
       id: 1,
       street: "123 street name rd",
       city: "New York"
      }
    }
  ]
}
```

And if you wanted to show a list of the person's cities, with it's ID, you'd need to tell the component how to pick each of these things out.

```js
{
  record: ["people"],
  label: ["address", "city],
  value: ["address", "id"]
}
```

This would show a list of the city names, and their value when selected would be the city's ID. The ID would get passed to be used in the next level down in the tree.

**Comment by kevde** [comment]
How can I determine the tree of the options via the API. Do you also have a sample API that can have the heirarchy?

**Comment by mattmonihan** [comment]
The user would know the API's design ahead of time, and those options would be included as part of the `"ui:options"` object. The point of this task is to have the component take the object with the key `remote` from the example above and pull the api from the `url` field and then, based on what's picked, use that value to populate `{{parent}}` in the second level of the tree.

**Pull request merged**